src/transformers/trainer.py

```diff
@@ -1588,12 +1591,22 @@ def _inner_training_loop(
             )
 
         # Compute absolute values for logging, eval, and save if given as ratio
-        if args.logging_steps and args.logging_steps < 1:
-            args.logging_steps = math.ceil(max_steps * args.logging_steps)
-        if args.eval_steps and args.eval_steps < 1:
-            args.eval_steps = math.ceil(max_steps * args.eval_steps)
-        if args.save_steps and args.save_steps < 1:
-            args.save_steps = math.ceil(max_steps * args.save_steps)
+        # Also stores the ratios in `self._step_ratio` on the first pass
+        if args.logging_steps:
```

Comment:
Let's also change to test for a real bool instead of relying on Python magic conversion. (Here and below)